Source/Core/VideoCommon/IndexGenerator.cpp

```diff
@@ -246,10 +326,13 @@ void IndexGenerator::AddExternalIndices(const u16* indices, u32 num_indices, u32
   m_base_index += num_vertices;
 }
 
-u32 IndexGenerator::GetRemainingIndices() const
+u32 IndexGenerator::GetRemainingIndices(OpcodeDecoder::Primitive primitive) const
 {
   // -1 is reserved for primitive restart (OGL + DX11)
-  constexpr u32 max_index = 65534;
+  u32 max_index = 65534;
+
+  if (g_Config.UseVSForLinePointExpand() && primitive >= OpcodeDecoder::Primitive::GX_DRAW_LINES)
+    max_index = (max_index >> 2) - 1;  // -1 for primitive restart
 
   return max_index - m_base_index;
```

Comment:
Probably would be better to write it like this:

```c++
  u32 max_index = USHRT_MAX;

  if (g_Config.UseVSForLinePointExpand() && primitive >= OpcodeDecoder::Primitive::GX_DRAW_LINES)
    max_index = (max_index >> 2);

  // The highest value (-1) is reserved for primitive restart (OGL + DX11)

  return max_index - m_base_index;
```

I'm also not 100% sure that the "OGL + DX11" part is correct - are those the only backends that actually use primitive restart, or is that comment just old enough that those were the only backends at the time?

Reply:
All renderers use it now, the comment is probably just old